engine/tests/phpunit/Elgg/FormsServiceTest.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Elgg;
+
+/**
+ * @group FormsService
+ */
+class FormsServiceTest extends \PHPUnit_Framework_TestCase {
+
+	/**
+	 * @var ViewsService
+	 */
+	protected $views;
+
+	/**
+	 * @var string
+	 */
+	protected $viewsDir;
+
+	public function setUp() {
+		$this->viewsDir = dirname(dirname(__FILE__)) . "/test_files/views";
+
+		$this->hooks = new PluginHooksService();
+		$logger = $this->getMock('\Elgg\Logger', array(), array(), '', false);
+
+		$this->views = new ViewsService($this->hooks, $logger);
+		$this->views->autoregisterViews('', "$this->viewsDir/default", 'default');
+
+		_elgg_services()->setValue('views', $this->views);
+		_elgg_services()->setValue('hooks', $this->hooks);
+		_elgg_services()->setValue('logger', $logger);
+	}
+
+	public function testCanRenderForm() {
+
+		$expected = elgg_view('forms/foo/bar.html');
+		$actual = elgg_view_form('foo/bar', [
+			'class' => 'foo-bar',
+		], [
+			'baz2' => 'bar2',
+		]);
+
+		$this->assertNotEmpty($expected);
+		$this->assertNotEmpty($actual);
+		$this->assertXMLStringEqualsXMLString($expected, $actual);
```

Comment:
@mrclay What's the best way to compare two XML strings ignoring new lines and tabs?